tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -48,12 +57,71 @@ class Port(Box):
     )
 
 
+class LumpedPort(Box):
+    """Lumped port source."""
+
+    name: str = pd.Field(
+        ...,
+        title="Name",
+        description="Unique name for the port.",
+        min_length=1,
+    )
+
+    voltage_axis: Axis = pd.Field(
+        ...,
+        title="Voltage Integration Axis",
+        description="Specifies the axis along which the E-field line integral is performed when "
+        "computing the port voltage. The integration axis must lie in the plane of the port.",
+    )
+
+    impedance: Complex = pd.Field(
+        50,
+        title="Reference impedance",
+        description="Reference port impedance for scattering parameter computation.",
+        units=OHM,
+    )
+
+    refine_mesh: bool = pd.Field(
+        True,
+        title="Port mesh refinement",
+        description="If ``True``, enables mesh refinement in the port region to ensure that "
+        "the port voltage and current are computed accurately.",
+    )
+
+    num_grid_cells: pd.PositiveInt = pd.Field(
+        DEFAULT_PORT_NUM_CELLS,
+        title="Port grid cells",
+        description="Number of mesh grid cells associated with the port along each direction. "
+        "This is used only if ``refine_mesh`` is ``True``.",
+    )
+
+    _plane_validator = assert_plane()
+
+    @cached_property
+    def injection_axis(self):
+        """Injection axis of the port."""
+        return self.size.index(0.0)
```

Comment:
what about a 2D simulation where the lumped port becomes 1D?

Reply:
Yea, didn't consider that. Not sure how to best handle this. I guess a 2D simulation might make sense as long as the injection axis and voltage axis are the dimensions that are part of the simulation.

Reply:
I added a validator to `TerminalComponentModeler` to ensure that the underlying simulation is 3D. I can't think of an application at the moment where it would be useful to include 2D simulations.